subprojects/language-scala/src/main/java/org/gradle/language/scala/tasks/PlatformScalaCompile.java

```diff
@@ -29,6 +30,7 @@
 /**
  * A platform-aware Scala compile task.
  */
+@CacheableTask
 @Incubating
 public class PlatformScalaCompile extends AbstractScalaCompile {
```

Comment:
❌  As seen by the failing `compilation is cached if the build executed from a different directory`, it is not enough to only add `@CacheableTask` to `PlatformScalaCompile`. It is also necessary to add the override of `getSource` with the correct path sensitivity: https://github.com/gradle/gradle/blob/e6b2c9027abb4a087df09df8ab4a1e40fad56465/subprojects/scala/src/main/java/org/gradle/api/tasks/scala/ScalaCompile.java#L70-L77

Moreover, the Java version should be tracked for Scala compilation, too. For `ScalaCompile`, this is done here:
https://github.com/gradle/gradle/blob/e6b2c9027abb4a087df09df8ab4a1e40fad56465/subprojects/scala/src/main/java/org/gradle/api/tasks/scala/ScalaCompile.java#L95-L106

We should add a test for `PlatformScalaCompile`, too, similar to this: https://github.com/gradle/gradle/blob/c8f68813fb4c4320fc1757e9da5dbc5af1a6bbb4/subprojects/scala/src/integTest/groovy/org/gradle/scala/compile/UpToDateScalaCompileIntegrationTest.groovy#L66-L89

I suggest adding both to `AbstractScalaCompile`.

Reply:
Updated as suggested